uber_sante/models/appointment.py

```diff
@@ -3,43 +3,21 @@ class Appointment:
 	When the appointment gets checked out it becomes a Booking object"""
 
 
-	def __init__(self, date, patient_id, patient_name, doctor_id, doctor_name, start_time, end_time):
-		# the date of the appointment
-		self.date = date
+	def __init__(self, patient_id, patient_name, availability, booking_type):
```

Comment:
Change "availability" to "availability_id"

Reply:
My thinking behind putting an availability object instead of simply an availability_id is that when an appointment is made, we have all the information of the availability object at hand. So if we don't keep it and only keep the availability_id then when we view cart we would have to query the availability from the table (again) to be able to display this appointment information: that information that was necessary to make the appointment that we just threw away.

Reply:
OK, can you then add the two methods i described above?

Reply:
There are many interaction diagrams that depend on the previous implementation of appointment, and now that it is changed, those diagrams should also be changed. I also don't see the purpose of storing the patient name (why do we need it), and the booking_type (already inside of availability).

All the interaction diagrams that passed an availability_id to the appointment should now fetch an availability object from the availability service first, and then pass that

Reply:
Some of the methods will have to be changed with this change. We just have to make sure everyone has the correct change in order to limit the effects while merging.

Reply:
@aa-software2112 I stored the patient name and id for front end display purposes, I can remove them but then we are a left with an appointment class that only wraps an availability

Reply:
It works with the patient_id and availability, even if it’s only one attribute away from the availability, in terms of the domain model it makes sense.